### PR TITLE
Conv2D Implementation

### PR DESCRIPTION
**Pull request by DamRsn**
- Implementation of streaming 2D convolution with Eigen (templated and non-templated)
  - In Tensorflow, the dimension should follow N x T x F x C (Batch, Time, Feature, Channel)
  - Support for "same" and "valid" padding (works the same way as tensorflow)
  - Support for stride on feature axis
  - Support for dilation on time axis
 
- Implementation of Batchnorm 2D with Eigen. Works very similarly as batchnorm1d, but now the channels consist of more than 1 value. But the number of weights stays the same. 

- Minimal change to original API: 
  - `in_size` and `out_size` are still used but set as `num_filters_in * num_features_in` and `num_filters_out * num_features_out`

- Test coverage for new layers, similar as other tests (but slightly different)

- Support for plain activation layer (conflict with base branch)

### CODE REVIEW

File: python/model_utils.py [hunk 1]
```diff
@@ -30,8 +30,19 @@ def get_layer_type(layer):
         if isinstance(layer, keras.layers.PReLU):
             return 'prelu'
 
+        if isinstance(layer, keras.layers.ReLU):
```
Comment (DamRsn) [inline python/model_utils.py:33]: To remove

Comment (DamRsn) [inline python/model_utils.py:33]: Remove by merging with base branch

File: RTNeural/conv2d/conv2d_eigen.h [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+#include "../common.h"
+#include "../conv1d_stateless/conv1d_stateless.h"
+#include <Eigen/Dense>
+
+namespace RTNeural
+{
+/**
+ * Dynamic implementation of a 2-dimensional convolution layer with no activation.
+ *
+ * @tparam T Type of the layer (float, double, int ...)
```
Comment (DamRsn) [inline RTNeural/conv2d/conv2d_eigen.h:14]: remove doc here

File: RTNeural/conv1d_stateless/conv1d_stateless.tpp [hunk 3]
```diff
@@ -0,0 +1,7 @@
+//
```
Comment (DamRsn) [inline RTNeural/conv1d_stateless/conv1d_stateless.tpp]: Remove header
